zospy/analyses/new/physicaloptics/physical_optics_propagation.py

```diff
@@ -0,0 +1,548 @@
+        """
+        super().__init__(settings or PhysicalOpticsPropagationSettings(), locals())
+
+    def run_analysis(self) -> DataFrame:
+        """Run the Physical Optics Propagation analysis."""
+        if self.settings.start_surface == "Ent. Pupil":
+            self.analysis.Settings.StartSurface.SetSurfaceNumber(0)
+        elif isinstance(self.settings.start_surface, int):
+            self.analysis.Settings.StartSurface.SetSurfaceNumber(self.settings.start_surface)
+        else:
+            raise ValueError(
+                f'start_surface value should be "Ent. Pupil" or an integer, got {self.settings.start_surface}'
+            )
+
+        if self.settings.end_surface == "Image":
+            self.analysis.Settings.EndSurface.UseImageSurface()
+        elif isinstance(self.settings.end_surface, int):
+            self.analysis.Settings.EndSurface.SetSurfaceNumber(self.settings.end_surface)
+        else:
+            raise ValueError(f'end_surface value should be "Image" or an integer, got {self.settings.end_surface}')
+
+        self.analysis.wavelength = self.settings.wavelength
+        self.analysis.field = self.settings.field
+        self.analysis.Settings.SurfaceToBeam = self.settings.surface_to_beam
+        self.analysis.Settings.UsePolarization = self.settings.use_polarization
+        self.analysis.Settings.SeparateXY = self.settings.separate_xy
+        self.analysis.Settings.UseDiskStorage = self.settings.use_disk_storage
+
+        # Beam settings
+        self.analysis.Settings.BeamType = constants.process_constant(
+            constants.Analysis.PhysicalOptics.POPBeamTypes, self.settings.beam_type
+        )
+
+        x_sampling = (
+            f"{self.settings.x_sampling}x{self.settings.y_sampling}"
+            if isinstance(self.settings.x_sampling, int)
+            else self.settings.x_sampling
+        )
+        y_sampling = (
+            f"{self.settings.x_sampling}x{self.settings.y_sampling}"
+            if isinstance(self.settings.y_sampling, int)
+            else self.settings.y_sampling
+        )
+
+        self.analysis.Settings.XSampling = constants.process_constant(
+            constants.Analysis.SampleSizes, standardize_sampling(x_sampling)
+        )
+        self.analysis.Settings.YSampling = constants.process_constant(
+            constants.Analysis.SampleSizes, standardize_sampling(y_sampling)
+        )
+        self.analysis.Settings.XWidth = self.settings.x_width
+        self.analysis.Settings.YWidth = self.settings.y_width
+
+        if self.settings.use_total_power:
+            self.analysis.Settings.UseTotalPower = True
+            self.analysis.Settings.TotalPower = self.settings.total_power
+        else:
+            self.analysis.Settings.UsePeakIrradiance = True
+            self.analysis.Settings.PeakIrradiance = self.settings.peak_irradiance
+
+        if self.settings.beam_file != "":
+            self.analysis.Settings.BeamFile = self.settings.beam_file
```

Comment:
@LucVV can you please check if `BeamFile` is indeed the name used in your OpticStudio version? 24R2 has `BeamTypeFilename`.